services/security-headers/security-headers.service.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const { BaseService } = require('..')
+
+const documentation = `
+<p>
+  The <a href="https://securityheaders.com/">Security Headers</a>
+  provide a easy mechanism to analyze HTTP response headers and give information on how to deploy missing headers.
+</p>
+</p>
+  The scan result will be hidden from the public result list and follow redirects will be on too.
+<p>
+`
+
+module.exports = class SecurityHeaders extends BaseService {
+  static get category() {
+    return 'monitoring'
+  }
+
+  static get route() {
+    return {
+      base: 'security-headers',
+      pattern: ':protocol(https|http)/:host',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Security Headers',
+        namedParams: {
+          protocol: 'https',
+          host: 'shields.io',
+        },
+        staticPreview: this.render({
+          grade: 'A+',
+        }),
+        documentation,
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'securityheaders',
+    }
+  }
+
+  static render({ grade }) {
+    if (!grade) {
+      return {
+        message: 'error',
+        color: 'lightgrey',
+      }
+    }
+
+    const colorMap = {
+      'A+': 'brightgreen',
+      A: 'green',
+      B: 'yellow',
+      C: 'yellow',
+      D: 'orange',
+      E: 'orange',
+      F: 'red',
+    }
+
+    return {
+      message: grade,
+      color: colorMap[grade],
+    }
+  }
+
+  async handle({ protocol, host }) {
+    let grade
+
+    try {
+      const { res } = await this._request({
+        url: `https://securityheaders.com/?q=${protocol}%3A%2F%2F${host}&hide=on&followRedirects=on`,
```

Comment:
You can use `qs` options like what is done in the [Wercker service](https://github.com/badges/shields/blob/b3be4d94d5ef570b8daccfd088c343a958988843/services/wercker/wercker.service.js#L87). Like that, no issues with URL encoding!

Reply:
Changed to use `qs` options :)